llmfoundry/tokenizers/tiktoken.py

```diff
@@ -140,117 +193,53 @@ def get_vocab(self) -> Dict[str, int]:
         Note: This function does not work properly due to difference in assumptions between tiktoken and Hugging Face tokenizers.
         Most uses do not need to use get_vocab, so this is not a priority to fix.
         """
-        warnings.warn(
-            'get_vocab does not work properly with TiktokenTokenizerWrapper. Please do not rely on it being perfectly correct.'
-            +
-            ' It will be called once init just to get the size of the vocab inside the base class.'
```

Comment:
I think its actually basically correct now (or as correct as other HF `get_vocab`). The extra ids are now only added because the gpt4 tokenizer has "holes" (i.e. a gap in vocab indices). because of the way HF uses `get_vocab`, it can't have holes. But this is a much less serious problem for someone using this function.

Reply:
Ah, got it. Update the function comment that says it does not work properly?

Reply:
👍 updated